overlord/snapshotstate/backend/helpers.go

```diff
@@ -152,26 +152,43 @@ func allUsers() ([]*user.User, error) {
 	return users, nil
 }
 
+var (
+	sysGeteuid   = sys.Geteuid
+	execLookPath = exec.LookPath
+)
+
 // maybeRunuserCommand returns an exec.Cmd that will, if the current
 // effective user id is 0 and username is not "root", call runuser(1)
-// to change to the given username before running the given command.
+// (or sudo(8) if runuser(1) isn't available) to change to the given
+// username before running the given command.
 //
 // If username is "root", or the effective user id is 0, the given
 // command is passed directly to exec.Command.
 //
 // TODO: maybe move this to osutil
 func maybeRunuserCommand(username string, args ...string) *exec.Cmd {
-	if username == "root" || sys.Geteuid() != 0 {
-		// runuser only works for euid 0, and doesn't make sense for root
-		return exec.Command(args[0], args[1:]...)
+	if sysGeteuid() == 0 && username != "root" {
+		// runuser and sudo happen to work the same way in this case.
+		// The main reason to prefer runuser over sudo is that runuser
+		// is part of util-linux, which is considered essential,
+		// whereas sudo is an addon which could be removed.
+		for _, cmd := range []string{"runuser", "sudo"} {
```

Comment:
Should we log an error/warning when neither is found?

Reply:
yes